pantheon-bundle/src/main/java/com/redhat/pantheon/validation/validators/NotNullValidator.java

```diff
@@ -0,0 +1,68 @@
+package com.redhat.pantheon.validation.validators;
+
+import com.redhat.pantheon.validation.model.ErrorDetails;
+import com.redhat.pantheon.validation.model.Violations;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ *  This is a sample validator that
+ *  <p>
+ *      <ol>Provides logic to check passed object is null or not </ol>
+ *      <ol>Accepts the data via setter when injected</ol>
+ *      <ol>Accepts the data via constructor instantiated normally</ol>
+ *      <ol>Returns its unique name via getName()</ol>
+ *      <ol>Returns the constraint violations via {@see Violations} instance</ol>
+ *  </p>
+ */
+@Component( service = NotNullValidator.class,
+        property = {
+            Constants.SERVICE_DESCRIPTION + "=Provides validation services",
+            Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        }
+)
+public class NotNullValidator implements Validator {
```

Comment:
I imagine adding a few more properties (annotations?) here to somehow list what events this validator cares about, sort of like what we already have in `SlingModelAdaptorFactory`:

```
@Component(
        service = NotNullValidator.class,
        property = {
                EVENT_TYPE + "=com.redhat.pantheon.validation.events.Status_Update",
                EVENT_TYPE + "=com.redhat.pantheon.validation.events.New_Module",
                EVENT_TYPE + "=com.redhat.pantheon.validation.events.Metadata_Changed"
        }
)
```

I made that up and probably the syntax isn't the best, but something along those lines. I'm hoping you can be the OSGi expert and advise on how best to supply the information here.